api/tests/unit/integrations/lead_tracking/hubspot/test_unit_hubspot_lead_tracking.py

```diff
@@ -212,3 +214,61 @@ def test_hubspot_with_existing_contact_and_new_organisation(
     # further hubspot resources.
     mock_create_company.assert_not_called()
     mock_create_contact.assert_not_called()
+
+
+def test_update_company_active_subscription(
+    organisation: Organisation,
+    settings: SettingsWrapper,
+    mocker: MockerFixture,
+) -> None:
+    settings.ENABLE_HUBSPOT_LEAD_TRACKING = True
+    mock_update_company = mocker.patch(
+        "integrations.lead_tracking.hubspot.client.HubspotClient.update_company"
+    )
+    hubspot_id = "12345"
+    # Create an existing hubspot organisation to mimic a previous
+    # successful API call.
+    HubspotOrganisation.objects.create(
+        organisation=organisation,
+        hubspot_id=hubspot_id,
+    )
+
+    assert organisation.subscription.plan == "free"
+
+    # When
+    organisation.subscription.plan = "scale-up-v2"
+    organisation.subscription.save()
+
+    # Then
+    mock_update_company.assert_called_once_with(
+        active_subscription=organisation.subscription.plan,
+        hubspot_company_id=hubspot_id,
+    )
```

Comment:
I'm _slightly_ concerned that I think this test relies on the `TASK_RUN_METHOD` environment variable being set to `SYNCHRONOUSLY`, right? 

I guess our options are: 

 1. force `settings.TASK_RUN_METHOD=TaskRunMethod.SYNCHRONOUSLY`
 2. Mock the task and assert task.delay is called, not the actual underlying hubspot client method (and then unit test the task function itself)
 3. something else?

Reply:
Well, I can force `settings.TASK_RUN_METHOD=TaskRunMethod.SYNCHRONOUSLY` if you think it's necessary, but many other tests I've written relied upon the implicit synchronicity but if you're concerned about it I can update the task to set it explicitly.

Reply:
Ok that's been updated. I also merged in the first of many merge conflicts from all this Hubspot work.